src/content/en/fundamentals/engage-and-retain/push-notifications/web-push-protocol.md

```diff
@@ -0,0 +1,613 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+
+{# wf_updated_on: 2017-03-03 #}
+{# wf_published_on: 2016-06-30 #}
+
+# The Web Push Protocol {: .page-title }
+
+{% include "web/_shared/contributors/mattgaunt.html" %}
+
+
+
+We've seen how a library can be used to trigger push messages, but what
+exactly are these libraries doing?
+
+Well, it's primarily making a network request, making sure it's the right format and sent the
+right way. The spec that defines this network request is the [Web Push
+Protocol](https://tools.ietf.org/html/draft-ietf-webpush-protocol).
+
+![Diagram of sending a push message from your server to a push
+service.](./images/svgs/server-to-push-service.svg)
+
+This section outlines how the server can identify itself with application
+server keys and how the encrypted payload and associated data is sent.
+
+This isn't a pretty side of web push and I'm no expert at encryption, but let's look through
+each piece since it's handy to know what these libraries are doing under the hood.
+
+## Application Server Keys
+
+When we subscribe a user, we pass in an `applicationServerKey`. This key is
+passed to the push service and used to check the application that subscribed
+the user is also the same application that is trigger push messages.
+
+When we trigger a push message, there are a set of headers that we send that
+allows the push service to authenticate the application (This is defined
+by the [VAPID spec](https://tools.ietf.org/html/draft-thomson-webpush-vapid)).
+
+What does all this actually mean and what exactly happens? Well these are the steps taken for
+application server authentication:
+
+1. The application signs some information with it's **private application key**.
+1. This signed information is sent to the push service as a header in the POST request.
+1. The push service uses the stored public key is received from `subscribe()` to check that the
+signed information is signed by the private key relating to the public key. *Remember*: The
+public key is the `applicationServerKey` passed into the subscribe call.
+1. If the signed information is valid the push service sends the push
+message to the user.
+
+An example of this flow of information is below (Not the legend in the bottom left to indicate
+public and private keys):
+
+![Illustration of how the private application server key is used when sending a
+message.](./images/svgs/application-server-key-send.svg)
+
+The "signed information" added to a header in the request is a JSON Web Token.
+
+### JSON Web Token
```

Comment:
Combination of what I've picked up from Chrome and people mixing up the terms between "push payload" and "JWT Payload", "POST Request Headers" and "JWT Headers".

This way - it's at least a little easier to seperate the concerns.